ray-operator/controllers/raycluster_controller.go

```diff
@@ -202,6 +202,75 @@ func (r *RayClusterReconciler) reconcileServices(instance *rayiov1alpha1.RayClus
 	return nil
 }
 
+func (r *RayClusterReconciler) needRemoveOldPod(template v1.PodTemplateSpec, pod v1.Pod) bool {
```

Comment:
this could be problematic, can we add some UTs for it?

Reply:
i added some extra unit tests.